### PR TITLE
feat: add bottom sheet + refacto existing modals

### PR DESCRIPTION
**Pull request by JohnBerd**
animatedModal

- bottom sheet context accessible in conversations
- deleted the emoji keyboard context
- a modal is accessible thanks to the hook useConversationModal

Reaction is not working anymore because of a node error of react-native-reanimated (wip)

### CODE REVIEW

**Review by aeddi — APPROVED** [review]
LGTM 👍

File: js/packages/components/chat/footer/ChatFooter.tsx [hunk 1]
```diff
@@ -56,24 +58,23 @@ export const ChatFooter: React.FC<ChatFooterProps> = React.memo(
...
 		const isFocused = useAppSelector(state => selectChatInputIsFocused(state, convPK))
-		const [addFileMenuVisible, setAddFileMenuVisible] = React.useState(false)
 
 		// computed
 		const isFake = !!(conversation as any)?.fake
 		const sendEnabled = !sending && !!(!isFake && (message || mediaCids.length > 0))
 		const horizontalGutter = 8 * scaleSize
-		const showQuickButtons =
-			!disabled && !sending && !message && !isFocused && mediaCids.length <= 0
+		const showQuickButtons = !disabled && !sending && !message && mediaCids.length <= 0
```
Comment (clegirar) [inline js/packages/components/chat/footer/ChatFooter.tsx:77]: we can use a useMemo here 👍

File: js/packages/components/chat/message/Reactions.tsx [hunk 2]
```diff
@@ -1,14 +1,16 @@
+import React, { createRef, FC, useCallback, useEffect, useState } from 'react'
+import { Text, View } from 'react-native'
+import { FlatList, ScrollView, TouchableOpacity } from 'react-native-gesture-handler'
```
[the remark this answers is not in the document]
Comment (JohnBerd) [reply js/packages/components/chat/message/Reactions.tsx:3]: I didn't know why but the onPress event weren't triggered when those were imported from react-native, I don't know why...

File: js/tsconfig.json [hunk 3]
```diff
@@ -16,10 +16,11 @@
 			"ES2015",
 			"DOM",
 			"ESNext.String"
-		] /* Specify library files to be included in the compilation. */,
-		"allowJs": true /* Allow javascript files to be compiled. */,
-		// "checkJs": true,                       /* Report errors in .js files. */
-		"jsx": "react-native" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
+			] /* Specify library files to be included in the compilation. */,
+			"skipLibCheck": true,
```
[the remark this answers is not in the document]
Comment (JohnBerd) [reply js/tsconfig.json:20]: After adding "react-native-reanimated", I figured out that tsc will also check in the node_modules when running 'make lint', this package is the only one having typescript errors I guess.
I tried to exclude node_modules in this file without any success.

If there are cleaner solutions for this, I can change it

**Comment by JohnBerd** [comment]
Moved to #3973

**Pull request closed without merging**